### DISCUSSION TITLE
Check how many features used in loss function

### DISCUSSION THREAD

**Discussion by nbjdttn** [MilesCranmer/PySR#532] [Q&A]
Hi,

Thank you for developing this library! I am wondering if there is a way to check how many features have been used in the current tree (except all the constants) when customising the loos functions? 

Thanks again!

**Comment by MilesCranmer** [comment]
Hi @nbjdttn,

Interesting question. I think you could do that like this:

```julia
# Create array to store whether feature was used:
was_feature_used = zeros(Bool, dataset.nfeatures)

# Loop through nodes in tree:
foreach(tree) do node
    if node.degree == 0 && !(node.constant)
        was_feature_used[node.feature] = true
    end
end

number_features_used = sum(was_feature_used)
```
